pkg/common/common_test.go

```diff
@@ -200,3 +200,127 @@ func TestGetImagePullPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateCliArgsFromConfigMap(t *testing.T) {
+	tests := []struct {
+		name          string
+		cliSubCommand []string
+		configMap     *corev1.ConfigMap
+		expectedArgs  []string
+	}{
+		{
+			name:          "Boolean argument variations",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"--default-snapshot-move":   "true",
+					"--another-key-mix-letters": "TrUe",
+					"key-no-prefix":             "False",
+					"string-not-bool":           "'False'",
+				},
+			},
+			expectedArgs: []string{
+				"server",
+				"--another-key-mix-letters=true",
+				"--default-snapshot-move=true",
+				"--key-no-prefix=false",
+				"--string-not-bool='False'",
+			},
+		},
+		{
+			name:          "All arguments with spaces, some without sigle quotes",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"default-volume-snapshot-locations": "aws:backups-primary, azure:backups-secondary",
+					"log-level":                         "'debug'",
+				},
+			},
+			expectedArgs: []string{
+				"server",
+				"--default-volume-snapshot-locations=aws:backups-primary, azure:backups-secondary",
+				"--log-level='debug'",
+			},
+		},
+		{
+			name:          "Preserve single and double '-' as key prefix",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"-default-volume-snapshot-locations": "aws:backups-primary, azure:backups-secondary",
+					"--log-level":                        "debug",
+					"-string-bool":                       "False",
```

Comment:
same as above

Reply:
fixed and added more options to cover single quotes.